core/src/repair/quic_endpoint.rs

```diff
@@ -0,0 +1,624 @@
+    config.alpn_protocols = vec![ALPN_REPAIR_PROTOCOL_ID.to_vec()];
+    let mut config = ServerConfig::with_crypto(Arc::new(config));
+    config
+        .transport_config(Arc::new(new_transport_config()))
+        .use_retry(true)
+        .migration(false);
+    Ok(config)
+}
+
+fn new_client_config(cert: Certificate, key: PrivateKey) -> Result<ClientConfig, rustls::Error> {
+    let mut config = rustls::ClientConfig::builder()
+        .with_safe_defaults()
+        .with_custom_certificate_verifier(Arc::new(SkipServerVerification {}))
+        .with_client_auth_cert(vec![cert], key)?;
+    config.enable_early_data = true;
+    config.alpn_protocols = vec![ALPN_REPAIR_PROTOCOL_ID.to_vec()];
+    let mut config = ClientConfig::new(Arc::new(config));
+    config.transport_config(Arc::new(new_transport_config()));
+    Ok(config)
+}
+
+fn new_transport_config() -> TransportConfig {
+    let mut config = TransportConfig::default();
+    config
+        .max_concurrent_bidi_streams(MAX_CONCURRENT_BIDI_STREAMS)
+        .max_concurrent_uni_streams(VarInt::from(0u8))
+        .datagram_receive_buffer_size(None);
+    config
+}
+
+async fn run_server(
+    endpoint: Endpoint,
+    remote_request_sender: Sender<RemoteRequest>,
+    cache: Arc<RwLock<ConnectionCache>>,
+) {
+    while let Some(connecting) = endpoint.accept().await {
+        tokio::task::spawn(handle_connecting_error(
+            endpoint.clone(),
+            connecting,
+            remote_request_sender.clone(),
+            cache.clone(),
+        ));
+    }
+}
+
+async fn run_client(
+    endpoint: Endpoint,
+    mut receiver: AsyncReceiver<LocalRequest>,
+    remote_request_sender: Sender<RemoteRequest>,
+    cache: Arc<RwLock<ConnectionCache>>,
+) {
+    while let Some(request) = receiver.recv().await {
+        tokio::task::spawn(send_request_task(
+            endpoint.clone(),
+            request,
+            remote_request_sender.clone(),
+            cache.clone(),
+        ));
+    }
+    close_quic_endpoint(&endpoint);
+}
+
+async fn handle_connecting_error(
+    endpoint: Endpoint,
+    connecting: Connecting,
+    remote_request_sender: Sender<RemoteRequest>,
+    cache: Arc<RwLock<ConnectionCache>>,
+) {
+    if let Err(err) = handle_connecting(endpoint, connecting, remote_request_sender, cache).await {
+        error!("handle_connecting: {err:?}");
+    }
+}
+
+async fn handle_connecting(
+    endpoint: Endpoint,
+    connecting: Connecting,
+    remote_request_sender: Sender<RemoteRequest>,
+    cache: Arc<RwLock<ConnectionCache>>,
+) -> Result<(), Error> {
+    let connection = connecting.await?;
+    let remote_address = connection.remote_address();
+    let remote_pubkey = get_remote_pubkey(&connection)?;
+    handle_connection_error(
+        endpoint,
+        remote_address,
+        remote_pubkey,
+        connection,
+        remote_request_sender,
+        cache,
+    )
+    .await;
+    Ok(())
+}
+
+async fn handle_connection_error(
```

Comment:
It is indeed intended to wrap `handle_connection` and handle resulting errors, e.g. by recording that into metrics.
It is not part of public api, and we can revise names later if needed.
For now I would rather use the same naming convention as in `turbine::quic_endpoint` in case we want to factor out the common parts later on.